tests/system_tests/test_core/test_data_types_full.py

```diff
@@ -130,3 +186,55 @@ def test_image_array_old_wandb_mp_warning(
         "Attempting to log a sequence of Image objects from multiple processes"
         " might result in data loss. Please upgrade your wandb server"
     )
+
+
+@pytest.mark.parametrize(
+    (
+        "create_media",
+        "file_ext",
+        "wandb_class",
+    ),
+    [
+        (
+            create_image,
+            "png",
+            wandb.Image,
+        ),
+        (
+            create_video,
+            "mp4",
+            wandb.Video,
+        ),
+        (
+            create_audio,
+            "wav",
+            wandb.Audio,
+        ),
+        (
+            create_bokeh,
+            "bokeh.json",
+            wandb.sdk.data_types.bokeh.Bokeh,
+        ),
+        (
+            create_object3d,
+            "pts.json",
+            wandb.Object3D,
+        ),
+        (
+            create_html,
+            "html",
+            wandb.Html,
+        ),
+    ],
+)
+def test_log_media_with_pathlib_path(
+    user, wandb_backend_spy, create_media, file_ext, wandb_class
+):
+    temp_dir = tempfile.mkdtemp()
```

Comment:
If you use the [tmp_path](https://docs.pytest.org/en/6.2.x/tmpdir.html) fixture, it will already be a `pathlib.Path` and it will be cleaned up properly at the end of the test.